Azure/communication-ui-library-ios
# [Refactor] Use async functions in the Calling SDK Components

## Purpose
Combine has been used for wrapping the calls to the underlying calling SDK, using `Future` to convert to a publisher.
These must then be consumed as such.

With Swift's latest concurrency tools, we can simplify this to use `async throwing` functions in a majority of places and set up the base for further asynchronous contexts.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Clone the repository
* Run the demo app and notice that there is no change in functionality.

## What to Check
Verify that the following are valid
* All unit tests pass
* Functionality is unaffected
